scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1038,6 +1041,48 @@
                               :number-unmatched (- (count offers) (count offers-scheduled))
                               :stats (offers->stats offers)}})
 
+          (let [unmatched-job-uuids
```

Comment:
Could you describe the algorithm used to determine this both high level and comments within here. In particular, I'm concerned if there are any leaks, we can OOM. I'd like to have a strong confidence that (in all paths) we prevent that.

Reply:
Yep, I added some comments above and throughout, let me know what you think.

Reply:
Love them. They're great! Thanks.